### ISSUE TITLE
Has anyone successfully run rod in AWS Lambda?

### ISSUE THREAD

**Issue by DiMiTriFrog** [go-rod/rod#456]
I would like to know if anyone has run the rod in Aws Lambda, because I configured a Lambda function with Python and are so troubles in the way for run correctly de function.

Anyone who has done it and can share their experience, configuration (e.g. chrome options (--single-process), that it needs to run well in Lambda or if anyone has tried running the headless-chromium binary with rod in Lambda.

Thanks!

**Comment by ysmood** [comment]
Need some time to investigate it.

**Comment by playniuniu** [comment]
AWS Lambda support custom docker now

You may try to use custom docker bundle chromium and go-rod

https://docs.aws.amazon.com/lambda/latest/dg/lambda-images.html

**Comment by ysmood** [comment]
I tried to use the image, seems like AWS lambda has a lot of limitations for a docker container, such as you can't use the root user, and memory size is very limited.

**Comment by DiMiTriFrog** [comment]
I finally managed to get both a go binary and a docker running on lambda. Now I have a problem that I am trying to figure out but there is no information about it, both those two functions that I have discussed and a third one that I made with python + arsenic, have periods where they stop working and periods where they work just fine. In local those periods do not work (even in local docker), but there is something in Lambda that does not end up working well, I have paid to the support to see if they can help me but so far I have not gotten an answer.

I share with all you my configuration Docker and files.

### Docker Go
**[hello-world.go](https://github.com/DiMiTriFrog/LambdaChromium/blob/main/hello-world.go)**

**[Dockerfile](https://github.com/DiMiTriFrog/LambdaChromium/blob/main/Dockerfile)**

**headless-chromium**
In the same folder of the docker I have the headless-chromium I use get from: 
https://github.com/adieuadieu/serverless-chrome/releases

**[go.mod](https://github.com/DiMiTriFrog/LambdaChromium/blob/main/go.mod)**

**[go.sum](https://github.com/DiMiTriFrog/LambdaChromium/blob/main/go.sum)**


### Direct Go binary
Now, for the direct Go Runtime I used a a Lambda Layer that contains the headless-chromium. For make the binary I use the follow command:
`GOARCH=amd64 GOOS=linux go build filename.go`

**[main.go](https://github.com/DiMiTriFrog/LambdaChromium/blob/main/main.go)** (I have configurated in Lambda main binary like the binary should been executed


Well,  this functions works perfectly (include other functions do the same but with python + other libraries like arsenic), but periodically Lambda shoot errors:

**An example of error:**

![error](https://user-images.githubusercontent.com/38754548/126128259-663e1a7d-6f33-4ac0-81d1-421550f1b96e.png)

**The % of errors)**
When the lambda functions is in hight period of errors, locally the functions works normally.
![percent](https://user-images.githubusercontent.com/38754548/126128449-7723f8df-dcc2-4ad8-8314-fcd2047d41b0.png)

Maybe the errors could be related to network saturation, maybe something specific needs to be configured in the VPC group.
By solving this we could run go-rod functions in lambda without any problem.

If you need help in any configuration of external IP, VPC, private-subred and public, or other tell me.

**Comment by DiMiTriFrog** [comment]
The premium support has answer me (really good answer) and at this moment we think my function is not closing the connections when the function finish the run. Any way to close browser / proxy connection etc?

**Comment by pthieu** [comment]
@DiMiTriFrog were you able to make any progress on this issue?

**Comment by YoungiiJC** [comment]
Here's a working example I spun up, if anyone needs some help: https://github.com/YoungiiJC/go-rod-aws-lambda

**Comment by ysmood** [comment]
@YoungiiJC This is awesome, thank you!

**Comment by pthieu** [comment]
@YoungiiJC awesome! I looked through your code, looks like you're using a docker container to host the function and run it on Lambda, are you also using ECS to host the container image?

**Comment by YoungiiJC** [comment]
@pthieu yup. I'm deploying via the [SAM cli](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-cli-command-reference-sam-deploy.html), which automatically pushes the image to ECR.

Behind the scenes SAM glues everything together declared in the [template.yaml](https://github.com/YoungiiJC/go-rod-aws-lambda/blob/main/template.yaml) file.

**Comment by pthieu** [comment]
@YoungiiJC solid.

In this case, because of ECR, there would be a cost incurred on the AWS account right?

**Comment by YoungiiJC** [comment]
@pthieu I suspect so

**Comment by ysmood** [comment]
@YoungiiJC I made a PR to slightly make it less resource consuming:

https://github.com/YoungiiJC/go-rod-aws-lambda/pull/1
